packages/host/app/components/operator-mode/stack-item.gts

```diff
@@ -288,147 +267,154 @@ export default class OperatorModeStackItem extends Component<Signature> {
       style={{this.styleForStackedCard}}
     >
       <CardContainer class={{cn 'card' edit=(eq @item.format 'edit')}}>
-        <Header
-          @title={{this.headerTitle}}
-          class={{cn 'header' header--icon-hovered=this.isHoverOnRealmIcon}}
-          {{on
-            'click'
-            (optional (if this.isBuried (fn @dismissStackedCardsAbove @index)))
-          }}
-          data-test-stack-card-header
+        <RealmInfoProvider
+          @realmURL={{this.card.id}}
+          @returnUnknownRealmIfError={{true}}
         >
-          <:icon>
-            {{#if this.headerIcon.URL}}
-              <img
-                class='header-icon'
-                src={{this.headerIcon.URL}}
-                data-test-boxel-header-icon={{this.headerIcon.URL}}
-                alt=''
-                role='presentation'
-                {{on 'mouseenter' this.headerIcon.onMouseEnter}}
-                {{on 'mouseleave' this.headerIcon.onMouseLeave}}
-              />
-            {{/if}}
-          </:icon>
-          <:actions>
-            {{#if (eq @item.format 'isolated')}}
-              <Tooltip @placement='top'>
-                <:trigger>
-                  <IconButton
-                    @icon='icon-pencil'
-                    @width='24px'
-                    @height='24px'
-                    class='icon-button'
-                    aria-label='Edit'
-                    {{on 'click' (fn @edit @item)}}
-                    data-test-edit-button
-                  />
-                </:trigger>
-                <:content>
-                  Edit
-                </:content>
-              </Tooltip>
-            {{else}}
-              <Tooltip @placement='top'>
-                <:trigger>
-                  <IconButton
-                    @icon='icon-pencil'
-                    @width='24px'
-                    @height='24px'
-                    class='icon-save'
-                    aria-label='Finish Editing'
-                    {{on 'click' (fn @save @item true)}}
-                    data-test-edit-button
-                  />
-                </:trigger>
-                <:content>
-                  Finish Editing
-                </:content>
-              </Tooltip>
-            {{/if}}
-            <div>
-              <BoxelDropdown>
-                <:trigger as |bindings|>
+          <:ready as |realmInfo|>
```

Comment:
what do we show when the `realmInfo` is being determined (since it's async)?

Reply:
Hmm, we don't display anything in the header. Do you think it would be better to revert to the old approach rather than using the `RealmInfoProvider` for this case?

Reply:
the old way was synchronous, yes? if so then yeah it's probably a better idea

Reply:
No, it's asynchronous as well https://github.com/cardstack/boxel/pull/700/files#diff-ca60a92fb9d0b21d7b89b8245f7a84b3078b4585c6b47ea474a536d7bcedd984L183. However, in the old way, we could still display the entire header, although without the right realm icon and realm name.

Reply:
Updated [f2df233](https://github.com/cardstack/boxel/pull/700/commits/f2df233217957ff8dec66b910b976725c8cc4eff)